### ISSUE TITLE
[BUG] `animation` example failing

### ISSUE THREAD

**Issue by MalteT** [amethyst/amethyst#2129]
## Description
The `animation` example compiles and runs but continuesly logs the following error:
```
[ERROR][amethyst_animation::systems::control] Failed to acquire animated component.
Is the component you are trying to animate present on the target entity: Entity(1, Generation(1))
```
The same error occurs when running the example for tag `v0.13.2`.

## Reproduction Steps

1. Clone current master branch.
2. Compile and run the `animation` example with `vulkan` backend (--features vulkan)

## What You Expected to Happen

No logged errors and something animated in the opened window.

## What Actually Happened

Static image in the window and the above error in the console.

## Your Environment

1. Operating System: `64-bit Arch linux`
2. Operating System Version `rolling`
3. Version of Amethyst `master`

**Comment by starccy** [comment]
Same here. When I run this example, it even freezes my whole DE.

**Comment by WimbledonLabs** [comment]
I also saw this issue. I fixed it by adding `.with(RestState::new(Transform::default()))` to line 87 of `examples/animation/main.rs`.

`.with(Transform::default())` also works, but it's less explicit.

The documentation doesn't make it clear that this is required (and in fact, the book points to _this_ example for how to implement animations). I'm not actually certain why this is required (I _just_ started using Amethyst a couple hours ago). I just inferred that this was needed by looking at the `start_animation` function from `amethyst_animation/src/systems/control.rs` (where this error originates from).

It looks like the rest state is only used in `amethyst_animation/src/systems/sampling.rs` in `fn process_sampler`. Even there it's only used to support `EndControl::Normal`. To me it's a little weird to require an additional component for each thing that's animated even if `EndControl::Stay` would do the job just fine.

Anyways, thanks to everyone who's working on Amethyst. It looks neat!
